src/long.ts

```diff
@@ -246,13 +247,37 @@ export class Long extends BSONValue {
   }
 
   /**
+   * @internal
+   * Returns false for an string that contains invalid characters for its radix, else returns the original string.
+   * @param str - The textual representation of the Long
+   * @param radix - The radix in which the text is written (2-36), defaults to 10
+   */
+  private static validateStringCharacters(str: string, radix?: number): false | string {
+    radix = radix ?? 10;
+    const validCharacters = '0123456789abcdefghijklmnopqrstuvwxyz'.slice(0, radix);
+    // regex is case insensitive and checks that each character within the string is one of the validCharacters
+    const regex = new RegExp(`[^-+${validCharacters}]`, 'i');
+    return regex.test(str) ? false : str;
+  }
+
+  /**
+   * @internal
    * Returns a Long representation of the given string, written using the specified radix.
+   * Throws an error if `throwsError` is set to true and any of the following conditions are true:
+   *  - the string contains invalid characters for the given radix
+   *  - the string contains whitespace
    * @param str - The textual representation of the Long
+   * @param validateStringCharacters - Whether or not invalid characters should throw an error
    * @param unsigned - Whether unsigned or not, defaults to signed
    * @param radix - The radix in which the text is written (2-36), defaults to 10
    * @returns The corresponding Long value
    */
-  static fromString(str: string, unsigned?: boolean, radix?: number): Long {
+  private static _fromString(
+    str: string,
+    validateStringCharacters: boolean,
+    unsigned?: boolean,
+    radix?: number
+  ): Long {
     if (str.length === 0) throw new BSONError('empty string');
     if (str === 'NaN' || str === 'Infinity' || str === '+Infinity' || str === '-Infinity')
       return Long.ZERO;
```

Comment:
We can run every validation before _fromString is called but there's certain validations that `Long.fromString` and `Long.fromStringStrict` share (no empty string, no interior hyphen, no invalid radices) that would cause some code repetition. Which is why I've only moved the round trip check outside of the function